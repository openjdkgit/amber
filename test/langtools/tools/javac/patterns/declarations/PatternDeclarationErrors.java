import java.io.IOException;

/*
 * @test /nodynamiccopyright/
 * @summary Verify error related to annotations and patterns
 * @enablePreview
 * @compile/fail/ref=PatternDeclarationErrors.out -XDrawDiagnostics -XDdev --should-stop=at=FLOW PatternDeclarationErrors.java
 */
public class PatternDeclarationErrors {
    public class D {
        public pattern D(Object v1, Float out) {
            match D(10.0f); // not matching signature
        }

        public pattern D(Float out, Integer v1) {
            match D2(10.0f, 2); // no matching name of pattern declaration
        }

        public pattern D(Float out, Double v1) {
            match D(10.0f, "2"); // inconvertible types
        }

        public void D3(Float out, Integer v1) {
            match D3(10.0f, 2); // match in regular method
        }

        public pattern D(Float out, Character v1) {
            match D(10.0f, '2');
            System.out.println("unreachable"); // unreachable
        }
    }

    public class ExitErrors {
        public pattern ExitErrors(int out) {
           break; // no break
        }

        public pattern ExitErrors(float out) {
            return Integer.valueOf(42); // no return
        }

        public pattern ExitErrors(char out) {
            yield Integer.valueOf(42); // no yield
        }
    }

    public class ExceptionErrors {
        public pattern ExceptionErrors(int out) {
            throw new Error(); // no throws in patter declaration body
        }

        public pattern ExceptionErrors(float out) throws IOException { // no throws in pattern signature
            throw new IOException(); // no throws in patter declaration body
        }
    }

    public record R(int i) {
        public pattern R(int i) { match R(42); }
    }
}
