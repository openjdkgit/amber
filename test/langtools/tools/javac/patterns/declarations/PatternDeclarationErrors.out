PatternDeclarationErrors.java:12:13: compiler.err.match.not.matching.pattern.declaration.signature
PatternDeclarationErrors.java:16:13: compiler.err.match.pattern.name.wrong
PatternDeclarationErrors.java:20:28: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: java.lang.String, java.lang.Double)
PatternDeclarationErrors.java:24:13: compiler.err.match.outside.pattern.declaration
PatternDeclarationErrors.java:35:12: compiler.err.break.outside.switch.loop
PatternDeclarationErrors.java:39:13: compiler.err.ret.outside.meth
PatternDeclarationErrors.java:43:13: compiler.err.no.switch.expression
PatternDeclarationErrors.java:49:13: compiler.err.pattern.declaration.no.throws
PatternDeclarationErrors.java:52:24: compiler.err.pattern.declaration.cant.throw.exception
PatternDeclarationErrors.java:53:13: compiler.err.pattern.declaration.no.throws
PatternDeclarationErrors.java:58:24: compiler.err.invalid.canonical.deconstructor.in.record: R
PatternDeclarationErrors.java:29:13: compiler.err.unreachable.stmt
- compiler.note.preview.filename: PatternDeclarationErrors.java, DEFAULT
- compiler.note.preview.recompile
12 errors